fuzz/fuzz_targets/fuzz_target_split_assemble.rs

```diff
@@ -0,0 +1,65 @@
+#![no_main]
+
+extern crate ctap2;
+extern crate libtock_drivers;
+#[macro_use]
+extern crate arrayref;
+
+use ctap2::ctap::hid::receive::MessageAssembler;
+use ctap2::ctap::hid::send::HidPacketIterator;
+use ctap2::ctap::hid::{HidPacket, Message};
+use libfuzzer_sys::fuzz_target;
+use libtock_drivers::timer::Timestamp;
+
+const DUMMY_TIMESTAMP: Timestamp<isize> = Timestamp::from_ms(0);
+const PACKET_TYPE_MASK: u8 = 0x80;
+
+// Converts a byte slice into Message
+fn raw_to_message(data: &[u8], len: usize) -> Message {
+    if len <= 4 {
+        let mut cid = [0; 4];
+        cid[..len].copy_from_slice(data);
+        Message {
+            cid,
+            cmd: 0,
+            payload: vec![],
+        }
+    } else {
+        Message {
+            cid: array_ref!(data, 0, 4).clone(),
+            cmd: data[4],
+            payload: data[5..].to_vec(),
+        }
+    }
+}
+
+/* Fuzzing HID packets splitting and assembling functions*/
+fuzz_target!(|data: &[u8]| {
+    let Message {
+        cid,
+        mut cmd,
+        payload,
+    } = raw_to_message(data, data.len());
```

Comment:
This is probably just style, but it removes a bit of the clutter. What about `let mut message = raw_to_message(data);` and then `message.clone()` and `message.cmd &= !PACKET_TYPE_MASK`?

Reply:
Ahh I can't believe it :D I'm too dumb. I had to break it down into parts because I got an error saying Message didn't derive some trait, probably I tried .copy() instead of .clone() first.